pyriemann/embedding.py

```diff
@@ -438,28 +446,344 @@ def locally_linear_embedding(
+            else:
+                # Pick initial step size based on where we were last time and
+                # look a bit further
+                # See Boumal, 2023, Section 4.3 for more insights.
+                alpha = 4 * (self.loss_evolution[-2] - loss) / \
+                        norm_direction**2
+
+            tau = 0.50
+            r = 1e-4
+            maxiter_linesearch = 25
+
+            retracted = retraction(current_sol, -alpha * direction)
+            Q_retract, Dsq_retract = self._compute_low_affinities(retracted)
+            loss_retracted = self._cost(P, Q_retract)
+
+            # Backtrack while the Armijo criterion is not satisfied
+            for _ in range(maxiter_linesearch):
+                if loss - loss_retracted > r * alpha * norm_direction**2:
+                    break
+                alpha = tau * alpha
+
+                retracted = retraction(current_sol, -alpha * direction)
+                Q_retract, Dsq_retract = self._compute_low_affinities(
+                    retracted)
+                loss_retracted = self._cost(P, Q_retract)
+            else:
+                warnings.warn("Maximum iteration in linesearched reached.")
+
+            # update variable for next iteration
+            current_sol = retracted
+
+            # test if the step size is small
+            crit = norm(current_sol, -alpha * direction)
+            if crit <= tol_step:
+                if self.verbosity >= 1:
+                    print("Min stepsize reached")
+                break
+
+            # test if the maximum time has been reached
+            if time() - initial_time >= self.max_time:
+                warnings.warn(
+                    "Time limite reached after " + str(i) + " iterations."
+                    )
+                break
+
+        else:
+            warnings.warn("Maximum iterations reached.")
+        if self.verbosity >= 1:
+            print(
+                "Optimization done in {:.2f} seconds.".format(
+                    time() - initial_time
+                    )
+                )
+        return current_sol
+
+    def fit(self, X, y=None):
+        """Fit Riemannian t-SNE.
+
+        Parameters
+        ----------
+        X : array_like of shape (n_matrices, n_channels, n_channels)
+            Set of SPD/HPD matrices.
+        y : None
+            Not used, here for compatibility with sklearn API.
+
+        Returns
+        -------
+        self : TSNE instance
+            The TSNE instance.
+        """
+        n_matrices, _, _ = X.shape
+
+        if self.perplexity is None:
+            self.perplexity = int(0.75 * n_matrices)
+
+        # Compute similarities in the high dimension space
+        P = self._compute_similarties(X)
+
+        # Sample initial solution close to the identity
+        sigma = 1
+        self.initial_point = sample_gaussian_spd(
+            n_matrices,
+            mean=np.eye(self.n_components),
+            sigma=sigma,
```

Comment:
`sigma` is not re-used
```suggestion
        self.initial_point = sample_gaussian_spd(
            n_matrices,
            mean=np.eye(self.n_components),
            sigma=1,
```